lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -1230,13 +1230,13 @@ void TFGraphLowering::addTPUEnqueueOp(BuiltinInst *inst, bool isInfeed,
     if (isInfeed)
       internalError(
           getUserSourceLocation(inst->getDebugLocation()),
-          "TPU infeed enqueue support enqueuing a single tensor -- did "
+          "TPU infeed enqueue supports enqueuing a single tensor -- did "
           "you specify shape?",
           diag::tfop_invalid_tfop);
     else
       internalError(
           getUserSourceLocation(inst->getDebugLocation()),
-          "TPU outfeed enqueue support enqueuing a single tensor -- did "
+          "TPU outfeed enqueue supports enqueuing a single tensor -- did "
```

Comment:
Random question, but why does outfeed need a shape?  It has an operand which should have a shape, can't it get it from there?

Reply:
Indeed this is not required for infeed enqueue, and I removed the TFLowerGraph logic for its shapes attr. For outfeed enqueue, there is not even a shapes attr (https://b.corp.google.com/issues/110538524).